editor-tool-autocomplete/editor-tool-autocomplete-api/src/main/java/org/xwiki/editor/tool/autocomplete/internal/AutoCompletionResource.java

```diff
@@ -353,11 +362,12 @@ private AutoCompletionMethodFinder getMethodFinder(String hint)
     protected VelocityContext getVelocityContext()
     {
         VelocityContext velocityContext = this.velocityManager.getVelocityContext();
+        XWikiContext context = this.xcontextProvider.get();
 
         // We add the "doc", "tdoc" and "sdoc" mappings since we don't get them from the Velocity Manager as they are
         // normally added based on the document passed in the request. However since we return the same method names
         // whatever the doc, we can manually add them.
-        XWikiDocument fakeDocument = createFakeXWikiDocument();
+        Document fakeDocument = createFakeXWikiDocument().newDocument(context);
```

Comment:
Indeed, I understand now. It wasn't immediately obvious to me when I reviewed it but it makes sense now and I now understand the issue. I think the title of the issue is way too generic, I'll fix it (that would have helped me understand the commit I think).